coordinator/src/services/translation/prescription/cancellation.ts

```diff
@@ -16,8 +16,14 @@ export function convertCancellation(
   const fhirFirstMedicationRequest = getMedicationRequests(fhirBundle)[0]
   const effectiveTime = extractEffectiveTime(fhirFirstMedicationRequest)
 
+  const messageId = getIdentifierValueForSystem(
```

Comment:
Seems like we have to do this twice, is there any way we can centralise this logic?

Reply:
Yep, changed